src/main/java/com/teragrep/pth_06/planner/walker/conditions/IndexStatementCondition.java

```diff
@@ -160,6 +156,8 @@ public boolean equals(final Object object) {
         if (object == null) return false;
         if (object.getClass() != this.getClass()) return false;
         final IndexStatementCondition cast = (IndexStatementCondition) object;
-        return this.condition().toString().equals(cast.condition().toString());
+        return this.value.equals(cast.value) &&
+                this.config.equals(cast.config) &&
+                this.tokenizer == cast.tokenizer;
```

Comment:
Here I'm unsure how to check the equality of the Tokenizer since it does not implement equals(Object o) method or allow method to check max token count. Now it checks that compared Tokenizers point to same instance